File: Domain-Module/src/main/kotlin/com/pawith/domain/auth/repository/OAuthRepository.java
package com.pawith.domain.auth.repository;


import com.pawith.domain.auth.entity.OAuth;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface OAuthRepository extends JpaRepository<OAuth, Long> {
    boolean existsBySub(String sub);

    boolean existsByUserId(Long userId);

    Optional<OAuth> findBySub(String sub);

    Optional<OAuth> findByUserId(Long userId);
}

File: Domain-Module/src/main/kotlin/com/pawith/domain/auth/repository/OAuthRepository.kt
package com.pawith.domain.auth.repository

import com.pawith.domain.auth.OAuth

interface OAuthRepository {
    fun existsBySub(sub: String)

    fun existsByUserId(userId: Long)

    fun findBySubOrNull(sub: String) : OAuth?

    fun findByUserIdOrNull(userId: Long) : OAuth?
}